### PR TITLE
MES-2938: Adding extra fields

### PR DESCRIPTION
**Pull request by szabolcs-keresztes**
adding special needs array
adding slot type

### CODE REVIEW

File: mes-test-schema/categories/B/index.json [hunk 1]
```diff
@@ -268,6 +273,10 @@
           "items": {
             "$ref": "#/definitions/initiator"
           }
+        },
+        "slotType": {
+          "description": "The type of the slot",
```
Comment (johnmorleybjss) [inline mes-test-schema/categories/B/index.json]: Not the most informative, could you relate it to some TARS domain concept?

Comment (szabolcs-keresztes) [reply mes-test-schema/categories/B/index.json]: True, but it doesn't exist in TARS. Not sure what else could I say. It's the type of the slot

File: mes-test-schema/categories/B/index.d.ts [hunk 2]
```diff
@@ -226,6 +226,10 @@ export interface TestSlotAttributes {
    * Whether the candidate has any special needs that require the D255 form to be completed
    */
   specialNeeds: boolean;
+  /**
+   * The special needs
+   */
+  specialNeedsArray?: any[];
```
Comment (ammarhaiderbjss) [inline mes-test-schema/categories/B/index.d.ts]: Think this should be `string[]`?

Perhaps try this - 

```
        "specialNeedsArray": {
          "description": "The special needs",
          "type": "array",
          "items": {
            "type": "string"
          }
        },
```

Comment (szabolcs-keresztes) [reply mes-test-schema/categories/B/index.d.ts]: Changed it and added the correct type: `string[]`

File: mes-test-schema/categories/B/index.json (hunk 1 above)
Comment (szabolcs-keresztes) [reply mes-test-schema/categories/B/index.json]: Added new description

**Pull request merged**